python/lsst/analysis/tools/tasks/sourceObjectTableAnalysis.py

```diff
@@ -20,19 +20,127 @@
 # along with this program.  If not, see <https://www.gnu.org/licenses/>.
 from __future__ import annotations
 
-__all__ = ("SourceObjectTableAnalysisConfig", "SourceObjectTableAnalysisTask")
+__all__ = (
+    "SourceObjectTableAnalysisConfig",
+    "SourceObjectTableAnalysisTask",
+    "ObjectEpochTableConfig",
+    "ObjectEpochTableTask",
+)
 
 import lsst.pex.config as pexConfig
+import lsst.pipe.base as pipeBase
 import numpy as np
 import pandas as pd
 from astropy.table import vstack
-from lsst.pipe.base import NoWorkFound
+from lsst.drp.tasks.gbdesAstrometricFit import calculate_apparent_motion
 from lsst.pipe.base import connectionTypes as ct
 from smatch import Matcher
 
 from ..interfaces import AnalysisBaseConfig, AnalysisBaseConnections, AnalysisPipelineTask
 
 
+class ObjectEpochTableConnections(
+    pipeBase.PipelineTaskConnections,
+    dimensions=("tract", "skymap"),
+    defaultTemplates={
+        # "inputName": "sourceTable_visit",
+        # "inputCoaddName": "deep",
+        # "associatedSourcesInputName": "isolated_star_sources",
+        # "outputName": "sourceObjectTable",
+    },
+):
+    refCat = ct.Input(
+        doc="Catalog of positions to use as reference.",
+        name="objectTable",
+        storageClass="DataFrame",
+        dimensions=["skymap", "tract", "patch"],
+        multiple=True,
```

Comment:
Why is this `multiple`? Oh, I think I see, this catalog is per-band? Please note that in the docstring.

Reply:
No, this is multiple patches.